gems/smithy/spec/interfaces/weld_spec.rb

```diff
@@ -43,7 +43,7 @@ def post_process(artifacts)
   # rubocop:enable Lint/UselessAssignment
 
   after(:all) do
-    SpecHelper.cleanup(['Weather'], @tmpdir)
+    SpecHelper.cleanup_generated(['Weather'], @tmpdir)
```

Comment:
This isn't directly related to this rectoring, but is something I've been thinking about - rather than needing to do this hacky before(:all)/after(:all) - we could have a with_generated method that would setup  a before and after block for us.

Reply:
I think shared context might work for that, but I don't like wrapping all of the tests in yet another block. I think it would be nice to collect all of the generated clients and handle them at clean up automatically, perhaps by having state in the helper.